### PR TITLE
blocks: set up auto-generated API docs

### PR DESCRIPTION
**Pull request by oandregal**
**Related to** https://github.com/WordPress/gutenberg/issues/14227

This PR sets up auto-generated API docs for the `blocks` package through the `npm run docs:build` command and updates its README file.

## Test instructions

- Delete or modify the contents at `packages/blocks/README.md` within the start `<!-- START TOKEN(Autogenerated API docs) -->` and end `<!-- END TOKEN(Autogenerated API docs) -->` tokens.
- Execute `npm run docs:build` and verify they are re-created.
- Review that the API docs match the JSDoc of the exported symbols.

### CODE REVIEW

File: packages/blocks/README.md [hunk 1]
```diff
@@ -151,67 +151,698 @@ _[(Example in ES2015+, JSX)](https://gist.github.com/aduth/fb1cc9a2296110a62b963
...
+    markup. See the [Element documentation](/packages/element/README.md) for more
+    information.
+-   Extracting `RandomImage` to a separate function allows us to reuse it in both
+    the editor-specific interface and the published content.
+-   The `edit` function should handle any case where an attribute is unset, as in
+    the case of the block being newly inserted.
+-   We only change the attributes of a block by calling the `setAttributes`
+    helper. Never assign a value on the attributes object directly.
+-   React provides conveniences for working with `select` element with
+    [`value` and `onChange` props](https://facebook.github.io/react/docs/forms.html#the-select-tag).
```
Comment (mkaz) [inline packages/blocks/README.md]: The list items have an extra space in front, which I don't think is needed. 
I'm more curious about how/why this change, I don't think it'll make a difference.

Comment (nosolosw) [reply packages/blocks/README.md]: Added it here https://github.com/WordPress/gutenberg/issues/14290 as a TODO item to review, this is something that can be improved in batch for all files once PRs are merged (to avoid rebasing nightmares).

This READMEs are read and then re-created by [the tool](https://github.com/WordPress/gutenberg/blob/master/packages/docgen/src/markdown/index.js#L27). My guess is that the library we use (`remark`) has some opinions about Markdown formatting.

Comment (mkaz) [reply packages/blocks/README.md]: Thanks for the explanation, makes sense

**Review by mkaz — APPROVED** [review]
Looks good :+1:
